tests/test_core.py

```diff
@@ -138,13 +151,7 @@ def _format_datetime_col(self):
 
 
 def test_init_empty_message(test_pd_df, caplog):
-    tdf = test_pd_df.copy()
-    tdf["extra_col"] = "test_val"
-    tdf["extra_col"] = np.nan
-
-    res = IamDataFrame(tdf)
-    assert "Primary Energy|Coal" not in res.variables().tolist()
-
+    IamDataFrame(data=df_empty)
     drop_message = (
         "Formatted data is empty! (perhaps there is a column full of nans?)"
```

Comment:
With this change, a column full of nan's would raise an error, so can you please change this message (and the test)
https://github.com/IAMconsortium/pyam/blob/7d46b2c1c3a46d799c1b09dbabc9dbe88e80ee07/pyam/utils.py#L259

to 'New IamDataFrame is empty!'

Similar to this warning message:
https://github.com/IAMconsortium/pyam/blob/7d46b2c1c3a46d799c1b09dbabc9dbe88e80ee07/pyam/core.py#L1124

Reply:
If the warning refers to IamDataFrame we'd have to make sure this function is only used when initializing IamDataFrame.
This is the case currently but it might change. If we are sure that it will not change then the `format_data` function should be converted to a method of `IamDataFrame`.

For now I just changed the warning message to `'Formatted data is empty!'` - I'm happy to further adapt the PR if you think `format_data` should be moved to `IamDataFrame`.

Reply:
That warning message is also fine, thanks!

In light of discussions in #335 and moving to `xarray` asbackend, there will be much more refactoring of this part anyway, so let's keep this PR nice and tidy.